### PR TITLE
VRF policy driven TE

### PR DESCRIPTION
**Pull request by xw-g**
Add test plans for VRF policy driven TE

### CODE REVIEW

File: feature/gribi/vrf_policy_driven_te/README.md [hunk 1]
```diff
@@ -0,0 +1,817 @@
...
+NH#1001 -> {
+  encapsulate_header: OPENCONFIGAFTTYPESENCAPSULATIONHEADERTYPE_IPV4
+  ip_in_ip {
+    dst_ip: "203.0.113.101"
+    src_ip: "ipv4_outer_src_222"
+  }
+  network_instance: "TE_VRF_222"
+}
+IPv4Entry {203.0.113.101/32 (TE_VRF_222)} -> NHG#9 (DEFAULT VRF) -> {
+  {NH#3, DEFAULT VRF, weight:1,ip_address=192.0.2.103},
```
Comment (arvbaska1) [inline feature/gribi/vrf_policy_driven_te/README.md:528]: Should this be different NH# , NH#3 is associated with 192.0.2.104 ?

File: feature/gribi/vrf_policy_driven_te/README.md [hunk 2]
```diff
@@ -0,0 +1,817 @@
...
+  backup_next_hop_group: 201 // decap to DEFAULT VRF
+}
+IPv4Entry {192.0.2.103/32 (DEFAULT VRF)} -> NHG#10 (DEFAULT VRF) -> {
```
Comment (arvbaska1) [inline feature/gribi/vrf_policy_driven_te/README.md:531]: 192.0.2.103/32 prefix has NH entry over Port5, so this should be another VIP ?

File: feature/gribi/vrf_policy_driven_te/README.md [hunk 3]
```diff
@@ -0,0 +1,817 @@
...
+  {NH#12, DEFAULT VRF, weight:1,mac_address:magic_mac, interface-ref:dut-port-7-interface},
+}
+
+```
+
+*   Install a BGP route resolved by ISIS in default VRF to rout traffic out of DUT port-8.
+
+*   Install an 0/0 static route in ENCAP_VRF_A and ENCAP_VRF_B pointing to the DEFAULT VRF.
```
Comment (arvbaska1) [inline feature/gribi/vrf_policy_driven_te/README.md:539]: The OC Local routing model doesn't look to be having a nexthop network-instance leaf.
So should this default static route in Encap VRF be part of the DUT startup baseconfig ?

File: feature/gribi/vrf_policy_driven_te/README.md [hunk 4]
```diff
@@ -0,0 +1,817 @@
...
+
+2.  Apply vrf selection policy `vrf_selection_policy_w` to DUT port-1.
+
+3.  Send the following 6in4 and 4in4 flows to DUT port-1:
+
+    ```
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_match`
+    * dscp: `dscp_encap_no_match`
+    * outter_src: `ipv4_outter_src_111`
```
Comment (singh-prem) [inline feature/gribi/vrf_policy_driven_te/README.md:656]: possible typo - used outter instead of outer

**Comment by xw-g** [comment]
Received approval from internal b/304783132. I'll merge this so we have a starting point.

@singh-prem and @arvbaska1, can you help raise separate PRs or bugs for your suggestion/questions? Thanks.

**Comment by arvbaska1** [comment]
> Received approval from internal b/304783132. I'll merge this so we have a starting point.
> 
> @singh-prem and @arvbaska1, can you help raise separate PRs or bugs for your suggestion/questions? Thanks.

Hi @xw-g , could you please just clarify this question here https://github.com/openconfig/featureprofiles/pull/2207/files#r1406820552 ? , i think also applies to this PR.

**Review by cfernanz — APPROVED** [review]

**Comment by xw-g** [comment]
> > Received approval from internal b/304783132. I'll merge this so we have a starting point.
> > @singh-prem and @arvbaska1, can you help raise separate PRs or bugs for your suggestion/questions? Thanks.
> 
> Hi @xw-g , could you please just clarify this question here https://github.com/openconfig/featureprofiles/pull/2207/files#r1406820552 ? , i think also applies to this PR.

https://github.com/openconfig/featureprofiles/pull/2514

**Pull request merged**